aten/src/ATen/native/README.md

```diff
@@ -305,6 +305,19 @@ There are also two special "generic" backends:
     DispatchStub should NOT be registered as CompositeExplicitAutograd, as
     DispatchStub only works for `CPU, CUDA`)
 
+  - `CompositeExplicitAutogradWithMutations`:
+    Similar to CompositeExplicitAutograd, but this key should be used if:
+    (1) Your kernel is written for an out-of-place operator
+    (2) *and* it calls internally into any mutation operators.
+    A common example of this type of kernel is for an out-of-place operator
+    that's implemented by creating an empty tensor, and calling into the corresponding out= operator.
+    We would like to distinguish between "ordinary" CompositeExplicitAutograd kernels
+    and these kernels, because some backends would not like
+    to decompose an out-of-place op into an inplace op.
+    LazyTensor + XLA are the two current examples of this - since they operate on a functional IR,
+    they would prefer to directly implement an out-of-place operator with their own kernel,
+    instead of using a decomposition that results in more mutation operators.
+
```

Comment:
I'm looking at the description here and feeling that this is all very complicated. I know I was the one who suggested composite with mutations in the first place (and though I have forgotten what more complicated solution we were replacing with this), and I think we should still put this PR in, but we should keep our eye out for a more general approach to this problem. Maybe the stuff on decompositions and primtorch feeds in here. cc @Chillee @mruberry